app/views/api/v2/job_invocations/main.json.rabl

```diff
@@ -21,6 +21,14 @@ child :targeting do
 
   child :hosts do
     extends 'api/v2/hosts/base'
+
+    if params[:host_status]
+      node(:job_status) do |host|
+        template_invocation = @template_invocations.find { |template_inv| template_inv.host_id == host.id }
```

Comment:
If you preprocess the data to have the right shape in the controller (say a hash mapping host ids to template invocations) then you can get instant access here instead of iterating through all of the template invocations (in the worst case). On my machine I was able to trim the request time from ~19 seconds to ~7